pkg/plugin/oauth2/setup.go

```diff
@@ -1,14 +1,27 @@
 package oauth2
 
 import (
+	"net/url"
+
+	"github.com/hellofresh/janus/pkg/config"
 	"github.com/hellofresh/janus/pkg/jwt"
-	"github.com/hellofresh/janus/pkg/oauth"
+	"github.com/hellofresh/janus/pkg/notifier"
 	"github.com/hellofresh/janus/pkg/plugin"
 	"github.com/hellofresh/janus/pkg/proxy"
+	"github.com/hellofresh/janus/pkg/router"
+	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	repo        Repository
+	adminRouter router.Router
+)
+
 func init() {
+	plugin.RegisterEventHook(plugin.StartupEvent, onStartup)
```

Comment:
Why should we? 
They are encapsulated on the same package. Can you give me one advantage of this design so I can better understand it?